### PR TITLE
Write failed findings to s3 bucket

### PR DESCRIPTION
**Pull request by danielmkn**
Add s3 bucket to the SAM template. 
Modify lambda policy, and add s3 bucket policy. 
Change `ComanyName` and `ProductName` to match AWS console UI.

### CODE REVIEW

File: template.yaml [hunk 1]
```diff
@@ -306,6 +306,20 @@ Resources:
...
+              - Effect: Allow
+                Action:
+                  - s3:Put*
+                  - s3:Get*
+                  - s3:List*
+                  - s3:Delete*
+                Resource:
+                  - !Join
+                    - ''
+                    - - 'arn:aws:s3:::'
```
Comment (alexhung) [inline template.yaml:321]: We should scope this to the AWS Account Id

File: template.yaml [hunk 2]
```diff
@@ -485,6 +500,51 @@ Resources:
...
+    UpdateReplacePolicy: Retain
+    Properties:
+      BucketName: !Sub "jfrog-xray-aws-failed-findings-${DeploymentEnvironment}-${AWS::Region}"
+      BucketEncryption:
+        ServerSideEncryptionConfiguration:
+          - ServerSideEncryptionByDefault:
+              SSEAlgorithm: AES256
+      LifecycleConfiguration:
+        Rules:
+          - Id: GlacierRule
```
Comment (alexhung) [inline template.yaml:515]: Let's name this correctly. Not glacier at all. This is rule for standard_ia storage class. Same with the `Prefix`.

File: template.yaml [hunk 3]
```diff
@@ -485,6 +500,51 @@ Resources:
...
+              - 's3:GetObject'
+              - 's3:PutObject'
+            Effect: Allow
+            Principal:
+              AWS:
+                - !GetAtt TransformerRole.Arn
+            Resource:
+              - !Join
+                - ''
+                - - 'arn:aws:s3:::'
```
Comment (alexhung) [inline template.yaml:544]: Same here, please scope this to the AWS Account Id

File: transformer/app.js [hunk 4]
```diff
@@ -190,6 +192,22 @@ const sendUpdateCommand = async (updateBody) => {
   };
 };
 
+const putFindingsIntos3bucket = async (failedFindings) => {
+  const bucketParams = {
+    Bucket: process.env.S3_BUCKET_NAME,
+    Key: `Failed_finding_${new Date().getTime().toString()}.json`,
```
Comment (alexhung) [inline transformer/app.js:198]: Let keep the file name casing consisten. Lowercase first letter please.

File: transformer/app.js [hunk 5]
```diff
@@ -190,6 +192,22 @@ const sendUpdateCommand = async (updateBody) => {
...
+    Body: JSON.stringify(failedFindings),
+  };
+  let data;
+  try {
+    data = await s3client.send(new PutObjectCommand(bucketParams));
+    logger.info(`Failed findings uploaded to s3://${bucketParams.bucketParams.Bucket}/${bucketParams.bucketParams.Key}`);
+  } catch (err) {
+    logger.debug('Error', { err });
```
Comment (alexhung) [inline transformer/app.js:206]: More descriptive error message please

**Review by alexhung — CHANGES_REQUESTED** [review]
A few minor changes needed.

File: template.yaml (hunk 2 above)
Comment (danielmkn) [reply template.yaml:515]: Leftovers, removed, thanks!

File: template.yaml (hunk 1 above)
Comment (danielmkn) [reply template.yaml:321]: It's already scoped to the s3 bucket name. Keeping this.

File: template.yaml (hunk 3 above)
Comment (danielmkn) [reply template.yaml:544]: It's already scoped to the s3 bucket name. Keeping this.

**Review by alexhung — APPROVED** [review]

**Pull request merged**